exception-handling/NullReferenceException/NullReferenceException/ExceptionMethods.cs

```diff
@@ -0,0 +1,139 @@
+﻿namespace NullReference
+{
+    public class ExceptionMethods
+    {
+        public List<string> StudentList() 
+        {
+            List<string> studentList = null;
+
+            studentList.Add("John Doe");
+
+            return studentList;
+        }
+
+        public List<string> FixedStudentList()
+        {
+            List<string> studentList = new List<string>();
+
+            studentList.Add("John Doe");
+
+            return studentList;
+        }
+
+        public int[] RandomNumbers()
+        {
+            var rand = new Random();
+
+            int[] numbers = null;
+
+            for (int i = 0; i < numbers.Length; i++) 
+            {
+                numbers[i] = rand.Next();
+            }
+         
+            return numbers;
+        }
+
+        public int[] FixedRandomNumbers()
+        {
+            var rand = new Random();
+
+            var numbers = new int[50];
+
+            for (int i = 0; i < numbers.Length; i++)
+            {
+                numbers[i] = rand.Next();
+            }
+
+            return numbers;
+        }
+
+        public string Teachers(string searchString)
+        {
+            var personObj = new Teacher();
+            
+            var people = personObj.AddRange(new String[] { "John", "Mary", "Jane", "Usher", "Andrew", "Grace", "Aston", "Sheila" });
```

Comment:
One last nitpick. Could we use `string` instead of `String` in all occurrences? There is a mix in the code. I have applied the changes to the article already 😄